### PR TITLE
Add search debounce option

### PR DESCRIPTION
**Pull request by fnkhz**
Required for implementing #2335
Requires PR: https://github.com/pbek/qmarkdowntextedit/pull/148

### CODE REVIEW

File: src/dialogs/settingsdialog.ui [hunk 1]
```diff
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>800</width>
-    <height>600</height>
+    <width>1045</width>
```
Comment (pbek) [inline src/dialogs/settingsdialog.ui:9]: Please don't change any width and height values. 😅

File: src/dialogs/settingsdialog.ui [hunk 2]
```diff
@@ -4753,8 +4753,8 @@ Just test yourself if you get sync conflicts and set a higher value if so.</stri
                 <string notr="true">&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
 &lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
 p, li { white-space: pre-wrap; }
-&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Noto Sans'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
-&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Sans Serif'; font-size:9pt;&quot;&gt;portable mode information will be loaded here&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
```
Comment (pbek) [inline src/dialogs/settingsdialog.ui:4756]: Please don't change any other text. 😁

File: src/mainwindow.cpp [hunk 3]
```diff
@@ -2739,6 +2739,9 @@ void MainWindow::readSettingsFromSettingsDialog(const bool isAppLaunch) {
             QStringLiteral("#eef067") : QStringLiteral("##141414")));
     }
 
+    const uint searchDebounceDelay = settings.value(QStringLiteral("searchDebounceDelay"), 0).toUInt();
+    ui->noteTextEdit->setDebounceDelay(searchDebounceDelay);
```
Comment (pbek) [inline src/mainwindow.cpp:2743]: ui->encryptedNoteTextEdit will need this too

File: src/mainwindow.cpp [hunk 4]
```diff
@@ -2739,6 +2739,9 @@ void MainWindow::readSettingsFromSettingsDialog(const bool isAppLaunch) {
             QStringLiteral("#eef067") : QStringLiteral("##141414")));
     }
 
+    const uint searchDebounceDelay = settings.value(QStringLiteral("searchDebounceDelay"), 0).toUInt();
```
Comment (pbek) [inline src/mainwindow.cpp:2742]: Don't forget that the delay in a QTimer is an int!

File: src/dialogs/settingsdialog.ui (hunk 1 above)
Comment (fnkhz) [reply src/dialogs/settingsdialog.ui:9]: Sorry, it was the editor :), will revert these.

File: src/mainwindow.cpp (hunk 4 above)
Comment (fnkhz) [reply src/mainwindow.cpp:2742]: static_casted the value to int in the call to `QTimer::setInterval`.

**Comment by pbek** [comment]
Since we will be setting the de-bounce value automatically I'll close this for now.

**Pull request closed without merging**